sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -42,30 +42,52 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
```

Comment:
We usually have `Async` suffix for async methods. Why is it async btw?

Reply:
I believe it's just how it was originally there. I don't see any reason of keeping it async. Our SDK doesn't make any external calls. They user of SDK could be doing that but we don't need to have this to be async.

@hakimms It's something we should definitely tackle when we move this to be an isolated process.

Reply:
Actually... i take it back, we have external call for TokenValidation, but happens before this. Regardless, this particular method doesn't need to be async

Reply:
Removed async for Failed and Completed.